src/storage/txn/actions/prewrite.rs

```diff
@@ -619,11 +619,9 @@ impl<'a> PrewriteMutation<'a> {
             lock.use_async_commit = true;
             lock.secondaries = secondary_keys.to_owned();
         } else if try_one_pc && lock.primary == self.key.to_raw()? {
-            // Set use_async_commit to true when try_one_pc and the key is primary. If
-            // use_async_commit is false and async-prewrite-apply is enabled, there is a
-            // chance that reads with max_ts cannot see the previous writes because the mem
-            // lock can be skipped and the data hasn't been applied yet.
-            lock.use_async_commit = true;
+            // Set `use_one_pc` to true when the key is primary. It's used to prevent the
+            // in-memory lock from being skipped when reading with max_ts.
+            lock.use_one_pc = true;
```

Comment:
Do we need to only set it for primary? Perhaps setting it to all locks is better?
...btw I just noticed that `lock.secondaries` seems can be set for no matter primary or secondaries... not sure if it's expected

Reply:
> Perhaps setting it to all locks is better?

Hmm... sounds reasonable.